src/bloqade/submission/ir/task_specification.py

```diff
@@ -51,6 +68,15 @@ def __hash__(self):
         return hash((RabiFrequencyAmplitude, self.global_))
 
     def discretize(self, task_capabilities: QuEraCapabilities):
+        """
+        Discretizes the global field values based on the task capabilities.
+
+        Args:
+           task_capabilities (QuEraCapabilities): Capabilities of the task for discretization.
```

Comment:
```suggestion
           task_capabilities (QuEraCapabilities): Capabilities of the target device
```

Sorry about this, the name of the argument isn't really intuitive here so this isn't really your fault but for context the capabilities for a particular quantum device not a task. This function is used to transform the IR nodes to be more compatible with a particular device.